src/api/spotPrice.ts

```diff
@@ -29,13 +29,14 @@ export const useSpotPrices = (
 ) => {
   const { tradeRouter } = useRpcProvider()
 
-  const assets = assetsIn
-    .filter((a): a is u32 | string => !!a)
-    .map((a) => a.toString())
+  const assets = new Set(
```

Comment:
I have just tested, that Set doesn't remove duplicates. But Map does

Reply:
There can't be duplicate values in Set